core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -471,6 +471,34 @@ public static SqlOperandTypeChecker variadic(
           .or(OperandTypes.family(SqlTypeFamily.MAP))
           .or(OperandTypes.family(SqlTypeFamily.ANY));
 
+  public static final SqlOperandTypeChecker STRING_OR_BINARY_ARRAY_STRING_OPTIONAL_STRING =
+      new FamilyOperandTypeChecker(
```

Comment:
personaly, the name STRING_OR_BINARY_ARRAY_STRING_OPTIONAL_STRING is not good. because the string contains BINARY family's type

Reply:
@rubenada it can not be. refers to the bigquery's doc https://cloud.google.com/bigquery/docs/reference/standard-sql/array_functions#array_to_string, the array's element type should be check in compiler process.

Reply:
@liuyongvs, If I understand correctly, what @rubenada meant was we should use `SqlTypeFamily.CHARACTER` instead of `SqlTypeFamily.STRING` for the second and third parameters. That make sense, I will updated it

Reply:
ok, i understand, thanks @zoudan